src/test/java/seedu/savenus/logic/parser/ParserUtilTest.java

```diff
@@ -167,4 +170,53 @@ public void parseTags_collectionWithValidTags_returnsTagSet() throws Exception {
 
         assertEquals(expectedTagSet, actualTagSet);
     }
+
+    @Test
+    public void parseLocations_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> ParserUtil.parseLocations(null));
+    }
+
+    @Test
+    public void parseLocations_emptyCollection_returnsEmptySet() throws Exception {
+        assertTrue(ParserUtil.parseLocations(Collections.emptyList()).isEmpty());
+    }
+
+    @Test
+    public void parseLocations_collectionWithInvalidLocations_throwsParseException() {
+        assertThrows(ParseException.class, () -> ParserUtil.parseTags(Arrays.asList(INVALID_LOCATION)));
+    }
+
+    @Test
+    public void parseCategories_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> ParserUtil.parseCategories(null));
+    }
+
+    @Test
+    public void parseCategories_emptyCollection_returnsEmptySet() throws Exception {
+        assertTrue(ParserUtil.parseCategories(Collections.emptyList()).isEmpty());
+    }
+
+    @Test
+    public void parseCategories_collectionWithInvalidLocations_throwsParseException() {
+        assertThrows(ParseException.class, () -> ParserUtil.parseCategories(Arrays.asList(INVALID_CATEGORY)));
+    }
+
+    @Test
+    public void parse_collectionWithValidTags_returnsTagSet() {
+        String validOpeningHours = "0000 0001";
+        String invalidOpeningHours = "abcd efgh";
+        try {
+            assertTrue(ParserUtil.parseOpeningHours(validOpeningHours).equals(new OpeningHours(validOpeningHours)));
+            assertTrue(ParserUtil.parseOpeningHours(invalidOpeningHours).equals(new OpeningHours(invalidOpeningHours)));
```

Comment:
i think you can use `assertParseFailure` for this!

Reply:
Will be fixed in the next commit!